File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l, r = 0, 0
        ans = 0
        count = {ch: 0 for ch in s}

        while r < len(s):
            if not count[s[r]]:
                count[s[r]] += 1
                ans = max(ans, (r - l) + 1)
                r += 1
            else:
                count[s[l]] -= 1
                l += 1

        return ans

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.cpp
class Solution {
public:
   
 int maxProfit(vector<int>& prices) {

        int l = 0, r = 0 , ans = 0;

        while(r<prices.size()){

               if (prices[l] >= prices[r])
                      l++;
               ans = max(ans, prices[r] - prices[l]);

               r++;
        }
        return ans;
 }

};
